lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -327,28 +328,23 @@ public function fetch($fetchMode = null, $cursorOrientation = PDO::FETCH_ORI_NEX
             throw new MysqliException($this->_stmt->error, $this->_stmt->sqlstate, $this->_stmt->errno);
         }
 
-        switch ($fetchMode) {
-            case FetchMode::NUMERIC:
-                return $values;
+        if ($fetchMode === FetchMode::NUMERIC) {
+            return $values;
+        }
 
+        assert(is_array($this->_columnNames));
```

Comment:
This is a good example (one of many) which shows that currently the `Statement` and `Result` concerns are intermixed in the same API just because of many underlying drivers do so. Column names belong to the result, which should be explicitly returned (not stored as a state) by the statement.